### PR TITLE
(Seeking feedback) load array columns as numpy array instead of list if use_numpy=True

### PR DESCRIPTION
**Pull request by edbordin**
Our clickhouse table has a column with the type `Array(Float32)` where each array has a length of about 10,000. I was interested in using the numpy support in clickhouse-driver to integrate efficiently with dask, but I found that array types are still loaded into a list, which is quite wasteful of memory compared to a numpy array.

After some hacking, these changes seem to work for my use-case, but I realise that there are probably a lot of edge cases with more complex/nested array types that I haven't tested. Using an array instead of a list may also have broken other things and would also possibly be considered a breaking change to the API for users.

Basically, I was wondering if you support something like this and whether you can think of a clean way to implement it properly? I'm happy to attempt implementing it myself given some guidance.

### CODE REVIEW

File: clickhouse_driver/columns/numpy/arraycolumn.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            nested_column = nested_column.nested_column
+
+        n_items = cur_level_slice_size if size > 0 else 0
+        nulls_map = None
+        if nested_column.nullable:
+            nulls_map = self._read_nulls_map(n_items, buf)
+
+        data = []
+        if n_items:
+            # use numpy array instead of list to improve memory usage
```
Comment (edbordin) [inline clickhouse_driver/columns/numpy/arraycolumn.py:33]: This is the only real change compared to `ArrayColumn._read()`

**Comment by xzkostyan** [comment]
It's good idea to use numpy arrays when we're dealing with `Array` column and using `Client` with `use_numpy` option.

As you can see `Array` type is stored as stripe of elements and offsets that forms items in result array.

Common `Array` tests (tests/columns/test_array.py) is a good start for numpy column tests. See `tests/numpy/columns` for numpy test examples.

You can create virtualenv, install packages from `setup.py:tests_require` and run all tests with `python setup.py test` locally. If you use ClickHouse in docker or non-standard port, tune `setup.cfg`

'Just working' Column code is a the good first step. The next step is it's time/memory optimization: `%prun` and `%memit` ipython's magics will be you friends it.

Feel free to ask any particular questions.

**Comment by xzkostyan** [comment]
Feel free to reopen PR when it will be ready.
